umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj

```diff
@@ -0,0 +1,42 @@
+(ns cmr.umm-spec.validation.temporal-extent
+  "Defines validations for UMM temporal extents."
+  (:require
+   [clj-time.core :as time]
+   [cmr.common.validations.core :as validations-core]
+   [cmr.umm-spec.date-util :as date]
+   [cmr.umm-spec.validation.umm-spec-validation-utils :as util]))
+
+(defn- range-date-time-validation
+  "Defines range-date-time validation"
+  [field-path value]
+  (let [{:keys [BeginningDateTime EndingDateTime]} value]
+    (when (and BeginningDateTime EndingDateTime (time/after? BeginningDateTime EndingDateTime))
+      {field-path [(format "BeginningDateTime [%s] must be no later than EndingDateTime [%s]"
+                           (str BeginningDateTime) (str EndingDateTime))]})))
+
+(defn- temporal-end-date-in-past-validator
+  "Validate that the end date in TemporalExtents is in the past"
+  [field-path value]
+  (when (and value (not (date/is-in-past? value)))
+    {field-path [(str "Ending date should be in the past. Either set ending date to a date "
+                      "in the past or remove end date and set the ends at present flag to true.")]}))
+
+(defn- ends-at-present-validation
+  "Validate that the collection ends at present flag and end date time are
+  not both set"
+  [field-path value]
+  (when (and (= true (:EndsAtPresentFlag value))
+             (seq (:RangeDateTimes value))
+             (not (some nil? (map :EndingDateTime (:RangeDateTimes value)))))
```

Comment:
Nit: Maybe easier to read as:
```(seq (keep :EndingDateTime (:RangeDateTimes value)))```
I just have trouble with double-negatives I think, nothing really wrong with the code.

Reply:
I don't think that is the same logic? I am looking to make sure that there is at least one date range with an ending date time of nil.

Reply:
Oh, I retract my comment and code.